src/gateway/environments/jsdom-sixteen/jsdom-sixteen-resource-loader.js

```diff
@@ -0,0 +1,144 @@
+// @flow
+import {URL} from "url";
+import {ResourceLoader} from "jsdom";
+import type {FetchOptions} from "jsdom";
+import type {RequestOptions, RenderAPI} from "../../types.js";
+import {getAgentForURL} from "../../../shared/index.js";
+import {
+    DefaultRequestOptions,
+    request,
+    abortInFlightRequests,
+} from "../../request.js";
+import {applyAbortablePromisesPatch} from "./apply-abortable-promises-patch.js";
+
+/**
+ * A ResourceLoader implementation for JSDOM sixteen-compatible versions of
+ * JSDOM that only allows for fetching JS files.
+ *
+ * A JS file request is identified by the regular expression:
+ *   /^.*\.js(?:\?.*)?/g
+ */
+export class JSDOMSixteenResourceLoader extends ResourceLoader {
```

Comment:
This is pretty much identical to the `CustomResourceLoader` of `react-render-server`, except for a couple of tweaks.